components/mnc-resolver/src/main/java/com/wso2telco/core/mnc/resolver/MNCQueryClient.java

```diff
@@ -75,36 +75,40 @@ public String QueryNetwork(String countryCode, String endUser) throws MobileNtEx
             }
         }
 
-        log.debug("MNCQueryClient : QueryNetwork : getStage :" + mcc.getStage());
-
-
-        if (mcc.getStage() == 0) {
-            //only one operator for the country
-            MobileNetwork = mcc.getDefaultMnc();
-        } else if (mcc.getStage() == 1) {
-            //1 range check only 
-            IProviderNetwork networkprovider = new MNCRangeCheck();
-            MobileNetwork = networkprovider.queryNetwork(String.valueOf(mcc.getCode()), endUser);
-        } else if (mcc.getStage() == 2) {
-            //path finder
-            IProviderNetwork networkprovider = new DNSSSLQueryClient();
-            String pfapiMnc = networkprovider.queryNetwork(String.valueOf(mcc.getCallingCode()), endUser.substring
-                    (mcc.getCallingCode().length()));
-            log.debug("MNCQueryClient : QueryNetwork : pfapiMnc :" + pfapiMnc);
-
-            if (pfapiMnc != null) {
-                MobileNetwork = McnRangeDbUtil.getMncBrand(String.valueOf(mcc.getCode()), pfapiMnc);
-            }
+        if (mcc != null) {
+            log.debug("MNCQueryClient : QueryNetwork : getStage :" + mcc.getStage());
+
 
+            if (mcc.getStage() == 0) {
+                //only one operator for the country
+                MobileNetwork = mcc.getDefaultMnc();
+            } else if (mcc.getStage() == 1) {
+                //1 range check only
+                IProviderNetwork networkprovider = new MNCRangeCheck();
+                MobileNetwork = networkprovider.queryNetwork(String.valueOf(mcc.getCode()), endUser);
+            } else if (mcc.getStage() == 2) {
+                //path finder
+                IProviderNetwork networkprovider = new DNSSSLQueryClient();
+                String pfapiMnc = networkprovider.queryNetwork(String.valueOf(mcc.getCallingCode()), endUser.substring
+                        (mcc.getCallingCode().length()));
+                log.debug("MNCQueryClient : QueryNetwork : pfapiMnc :" + pfapiMnc);
+
+                if (pfapiMnc != null) {
+                    MobileNetwork = McnRangeDbUtil.getMncBrand(String.valueOf(mcc.getCode()), pfapiMnc);
+                }
 
-        } else if (mcc.getStage() == 3) {
-            //range checker with fallback
-            IProviderNetwork networkprovider = new MNCRangeCheck();
-            MobileNetwork = networkprovider.queryNetwork(String.valueOf(mcc.getCode()), endUser);
 
-            if (MobileNetwork == null) {
+            } else if (mcc.getStage() == 3) {
+                //range checker with fallback
+                IProviderNetwork networkprovider = new MNCRangeCheck();
                 MobileNetwork = networkprovider.queryNetwork(String.valueOf(mcc.getCode()), endUser);
+
+                if (MobileNetwork == null) {
+                    MobileNetwork = networkprovider.queryNetwork(String.valueOf(mcc.getCode()), endUser);
+                }
             }
+        } else {
+            log.info("MCC is NULL");
         }
```

Comment:
Why did you introduce a info log here?

Reply:
with the previous code, if `mcc` is equal to `null`, there will be an exception on `mcc.getStage()`. But after adding null check, nothing will be printed on logs if `mcc` is equal to `null`. Because of that, I have added this extra info log.

Reply:
Ok then make it a debug log as all other logs are debug in this code.
log.debug("MNCQueryClient : QueryNetwork : MCC is NULL");

Reply:
> Ok then, won't it be better if we make it a debug log as all other logs are debug in this code.
> log.debug("MNCQueryClient : QueryNetwork : MCC is NULL");

done.